astropy/units/quantity.py

```diff
@@ -388,8 +388,7 @@ def __class_getitem__(cls, unit_shape_dtype):
 
         Notes
         -----
-        With Python 3.9+ or :mod:`typing_extensions`, |Quantity| types are also
-        static-type compatible.
+        |Quantity| types are also static-type compatible.
```

Comment:
In the pull request @nstarman opened to drop support for Python 3.9 he also found some code and documentation that should or could have been updated when support for Python 3.8 was dropped. That pull request has not been merged yet. Perhaps it would be a good idea to open a separate pull request for removing the obsolete mentions of Python 3.9 like this.

Reply:
You must be referring to #15603. I notice this specific line isn't touched there. I'm happy to open another PR with just this (and possibly other similar changes if I find any), but I'm also fine if @nstarman wants to add it to his PR instead !

Reply:
The problem with #15603 is that it seems to be stuck because the reference images for image comparison tests need to be regenerated. It would be better to split off things from there in new pull requests instead of adding more things to it that would then get stuck too.

Reply:
Agreed. Anything that can be paired off is good. Feel free to 🍒pick commits.

Reply:
Ok, I'll move it to a separate branch, and will open a PR in a few days with maybe more of the same !